index.ts

```diff
@@ -165,15 +223,18 @@ module.exports = class Plugin {
 
     if (this._readCompatError != null) throw this._readCompatError;
 
-    return this._readCompat.read(readTree);
+    return (this._readCompat as ReadCompat).read(readTree);
```

Comment:
rather then `as` can we guard against boolean?

Reply:
```ts
  if (this._readCompat) {
      return this._readCompat.read(readTree);
    }
```